lib/common/socket.c

```diff
@@ -245,6 +245,8 @@ static void dispose_ssl_output_buffer(struct st_h2o_socket_ssl_t *ssl)
 
     if (ssl->output.buf.capacity == h2o_socket_ssl_buffer_size)
         h2o_mem_free_recycle(&h2o_socket_ssl_buffer_allocator, ssl->output.buf.base);
+    else
+        free(ssl->output.buf.base);
```

Comment:
Per our coding guideline,
```suggestion
    if (ssl->output.buf.capacity == h2o_socket_ssl_buffer_size) {
        h2o_mem_free_recycle(&h2o_socket_ssl_buffer_allocator, ssl->output.buf.base);
    } else {
        free(ssl->output.buf.base);
    }
```